src/Microsoft.Performance.SDK.Runtime/CustomDataSourceReference.cs

```diff
@@ -42,13 +61,46 @@ protected CustomDataSourceReference(Type type)
         /// <param name="type">
         ///     The <see cref="Type"/> of instance being referenced by this instance.
         /// </param>
+        /// <param name="instanceFactory">
+        ///     The factory function for creating the instance.
+        /// </param>
         /// <exception cref="System.ArgumentNullException">
         ///     <paramref name="type"/> is <c>null</c>.
         /// </exception>
-        protected CustomDataSourceReference(Type type, Func<ICustomDataSource> instanceFactory)
+        public CustomDataSourceReference(
+            Type type,
+            Func<ICustomDataSource> instanceFactory,
+            CustomDataSourceAttribute metadata,
+            HashSet<DataSourceAttribute> dataSourceAttributes)
```

Comment:
copy/paste the doc comments from the other constructor

Reply:
Thanks